OpenRA.Mods.Common/Traits/Armament.cs

```diff
@@ -141,15 +147,20 @@ public Armament(Actor self, ArmamentInfo info)
 
 		public virtual WDist MaxRange()
 		{
-			return new WDist(Util.ApplyPercentageModifiers(Weapon.Range.Length, rangeModifiers));
+			return new WDist(Util.ApplyPercentageModifiers(Weapon.Range.Length, initialRangeModifiers));
```

Comment:
Don't these do different things? Previously `rangeModifiers` cached the traits but re-evaluated the query on `GetRangeModifier` each time. `initialRangeModifiers` here caches the results of the first call to `GetRangeModifier`. Is this an intentional change?